vuu-ui/packages/vuu-shell/src/layout-management/useLayoutManager.tsx

```diff
@@ -1,42 +1,41 @@
 import React, { useState, useCallback, useContext, useEffect } from "react";
-import { getLocalEntity, saveLocalEntity } from "@finos/vuu-filters";
-import { LayoutJSON } from "@finos/vuu-layout";
-import { getUniqueId } from "@finos/vuu-utils";
+import { getLocalEntity } from "@finos/vuu-filters";
+import { LayoutJSON, LayoutPersistenceManager } from "@finos/vuu-layout";
 import { LayoutMetadata, Layout } from "./layoutTypes";
 
 export const LayoutManagementContext = React.createContext<{
   layouts: Layout[],
   saveLayout: (n: Omit<LayoutMetadata, "id">) => void
 }>({ layouts: [], saveLayout: () => { } })
 
-export const LayoutManagementProvider = (props: { children: JSX.Element | JSX.Element[] }) => {
-
-  const [layouts, setLayouts] = useState<Layout[]>([])
+export const LayoutManagementProvider = (props: {
+    persistenceManager: LayoutPersistenceManager,
+    children: JSX.Element | JSX.Element[]
+  }) => {
+  const [layouts, setLayouts] = useState<Layout[]>([]);
+  const [tempLayouts, setTempLayouts] = useState<Layout[]>([]);
```

Comment:
tempLayout could be just a single layout

Reply:
type for it could be something like:  `Omit<Layout, "metadata"> & { tempMetadata: Omit<LayoutMetadata, "id"> }`